vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/post_claim_bitflyer/post_claim_bitflyer.cc

```diff
@@ -121,7 +148,14 @@ void PostClaimBitflyer::Request(const std::string& linking_info,
 void PostClaimBitflyer::OnRequest(const type::UrlResponse& response,
                                   PostClaimBitflyerCallback callback) {
   ledger::LogUrlResponse(__func__, response);
-  callback(CheckStatusCode(response.status_code));
+
+  auto result = CheckStatusCode(response.status_code);
+  if (result == type::Result::IN_PROGRESS) {
```

Comment:
What's the reason for choosing `IN_PROGRESS` here? Is the choice of `IN_PROGRESS` arbitrary?

Reply:
I chose `IN_PROGRESS` to indicate an intermediate result, actually to express the intention that the calculation of the result is `IN_PROGRESS`:
```
auto result = CheckStatusCode(response.status_code);
if (result == type::Result::IN_PROGRESS) {
  // disambiguating on the HTTP 403 (Forbidden)
  result = ParseBody(response.body);
}
```

Reply:
That makes sense, although it points to the limitations of using a general library-level error code for specific operations like `CheckStatusCode`: we're basically inventing a bespoke meaning for `IN_PROGRESS` because the operation may require more processing.

We might want to consider some other approaches. Perhaps `response.body` could be a parameter to `CheckStatusCode`, and you'd do the body parsing there if needed?

Reply:
Yeah, endpoint client implementations tend to follow the `CheckStatusCode + ParseBody` pattern, which might be a bit contrived here. I was also thinking of introducing a function like:
```
type::Result ProcessResponse(const type::UrlResponse& response) const;
```
and doing all the stuff in there.
How about that?

Reply:
Sounds great!